Apps/JobScheduler/src/Jobs/EmailJob.cs

```diff
@@ -137,21 +127,58 @@ private void ProcessEmails(IList<Email> resendEmails)
                 this.logger.LogInformation("Found {Count} emails to send", resendEmails.Count);
                 foreach (Email email in resendEmails)
                 {
-#pragma warning disable CA1031 //We want to catch exception.
                     try
                     {
-                        this.SendEmail(email);
+                        if (this.notifyConfiguration.Enabled)
+                        {
+                            this.SendEmailUsingNotify(email).GetAwaiter().GetResult();
+                        }
+                        else
+                        {
+                            this.SendEmail(email);
+                        }
                     }
                     catch (Exception e)
                     {
                         // log the exception as a warning but we can continue
                         this.logger.LogWarning("Error while sending {Id} - skipping for now\n{Exception}", email.Id.ToString(), e.ToString());
                     }
-#pragma warning restore CA1031 // Restore warnings.
                 }
             }
         }
 
+        private async Task SendEmailUsingNotify(Email email)
+        {
+            email.Attempts++;
+            try
+            {
+                _ = email.Template ?? throw new MissingFieldException("TemplateId is null");
+                EmailRequest emailRequest = new()
+                {
+                    EmailAddress = email.To!,
+                    TemplateId = this.notifyConfiguration.Templates[email.Template],
+                    Personalization = email.Personalization,
+                    Reference = email.Id.ToString("D"),
+                };
+                EmailResponse response = await this.notifyApi.SendEmail(emailRequest).ConfigureAwait(true);
+                email.NotificationId = response.Id;
+                email.EmailStatusCode = EmailStatus.Processed;
+                email.SentDateTime = DateTime.UtcNow;
+                this.emailDelegate.UpdateEmail(email);
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError(
+                    "Unexpected error while communicating to GC Notify API for email {Id}, Error = {Exception}",
+                    email.Id.ToString(),
+                    e.ToString());
+                email.LastRetryDateTime = DateTime.UtcNow;
+                email.EmailStatusCode = email.Attempts < this.maxRetries ? EmailStatus.Pending : EmailStatus.Error;
+                this.emailDelegate.UpdateEmail(email);
```

Comment:
This logic should probably go up - wouldn't this logic apply to SendEmail as well?